leaderboard/service/get_top_percentage_test.go

```diff
@@ -0,0 +1,178 @@
+package service_test
+
+import (
+	"context"
+	"errors"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/topfreegames/podium/leaderboard/database"
+	"github.com/topfreegames/podium/leaderboard/model"
+	"github.com/topfreegames/podium/leaderboard/service"
+)
+
+var _ = Describe("Service GetTopPercentage", func() {
+	var ctrl *gomock.Controller
+	var mock *database.MockDatabase
+	var svc *service.Service
+
+	var leaderboard string = "leaderboardTest"
+	var pageSize int = 10
+	var amount int = 30
+	var maxMembers int = 3
+	var order string = "asc"
+
+	membersReturnedByDatabase := []*database.Member{
+		{
+			Member: "member1",
+			Score:  float64(1),
+			Rank:   int64(0),
+		},
+		{
+			Member: "member2",
+			Score:  float64(2),
+			Rank:   int64(1),
+		},
+		{
+			Member: "member3",
+			Score:  float64(3),
+			Rank:   int64(2),
+		},
+	}
+
+	expectedMembersToReturn := []*model.Member{
+		{
+			PublicID: "member1",
+			Score:    1,
+			Rank:     1,
+		},
+		{
+			PublicID: "member2",
+			Score:    2,
+			Rank:     2,
+		},
+		{
+			PublicID: "member3",
+			Score:    3,
+			Rank:     3,
+		},
+	}
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		mock = database.NewMockDatabase(ctrl)
+
+		svc = &service.Service{mock}
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Describe("With order = desc", func() {
+		order = "desc"
+
+		It("Should return top percentage members if everything is OK", func() {
+			amount = 3
+			maxMembers = 10
+
+			mock.EXPECT().GetTotalMembers(gomock.Any(), gomock.Eq(leaderboard)).Return(100, nil)
+			mock.EXPECT().GetOrderedMembers(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(0), gomock.Eq(2), gomock.Eq(order)).Return(membersReturnedByDatabase, nil)
+
+			members, err := svc.GetTopPercentage(context.Background(), leaderboard, pageSize, amount, maxMembers, order)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(members).To(HaveLen(3))
+			Expect(members).To(Equal(expectedMembersToReturn))
+		})
+	})
+
+	Describe("With order = asc", func() {
+		order = "asc"
+
+		It("Should return bottom percentage members if everything is OK", func() {
```

Comment:
```suggestion
		It("Should return percentage members if everything is OK", func() {
```
I think that bottom suggests that it will get the last leaderboard members, but how the order is `asc`, it will get first members, with less score, this make the description a bit confuse.